test/e2e/util.go

```diff
@@ -1070,6 +1070,19 @@ func isInterconnectEnabled() bool {
 	return present && val == "true"
 }
 
+func singleNodePerZone() bool {
+	val, present := os.LookupEnv("KIND_NUM_NODES_PER_ZONE")
```

Comment:
nit: this might not be the best way to check, because it assumes KIND. I think we run some of these tests downstream for external gateway stuff. But this is something we can improve on later if we need to.

Reply:
should be more reliable now (and the previous version didn't work for non-ic)